### PR TITLE
fix(FEC-8411): add protection to validate external text tracks which are added without language

### PR DESCRIPTION
**Pull request by odedhutzler**
*added a check to see if there is no language, and trigger an error in that case
*refactor `getExternalTracks` function

### Description of the Changes

Please add a detailed description of the change, weather it's an enhancement or a bugfix.
If the PR is related to an open issue please link to it.

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment 
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/track/external-captions-handler.js [hunk 1]
```diff
@@ -116,37 +116,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
+  _maybeAddTrack(track: TextTrack, caption: PKExternalCaptionObject, playerTextTracks: Array<Track>, newTextTracks: Array<TextTrack>): void {
+    const sameLangTrack = playerTextTracks.find(textTrack => caption.language === textTrack.language);
+    if (!sameLangTrack) {
+      if (this._player.config.playback.useNativeTextTrack) {
+        this._addNativeTextTrack(track);
+      }
+      newTextTracks.push(track);
+      this._updateTextTracksModel(caption);
+    } else {
+      ExternalCaptionsHandler._logger.warn('duplicated language, taking the inbend option. Language: ', sameLangTrack.language);
```
Comment (OrenMe) [inline src/track/external-captions-handler.js]: typo - inband

File: src/track/external-captions-handler.js [hunk 2]
```diff
@@ -116,37 +116,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
-        url: caption.url,
-        type: caption.type
-      };
-      if (!sameLangTrack) {
-        if (this._player.config.playback.useNativeTextTrack) {
-          this._addNativeTextTrack(track);
-        }
-        newTextTracks.push(track);
+      if (!caption.language) {
+        const error = new Error(Error.Severity.RECOVERABLE, Error.Category.TEXT, Error.Code.UNKNOWN_LANGUAGE, caption);
```
Comment (dan-ziv) [inline src/track/external-captions-handler.js]: Add meaningful error data and not just the caption object. Use a key to describe the object.

File: src/track/external-captions-handler.js [hunk 3]
```diff
@@ -116,37 +116,76 @@ class ExternalCaptionsHandler extends FakeEventTarget {
...
+  /**
+   * checking if there is already a track with the same language
+   * @param {TextTrack} track - textTrack to be added text tracks array that will be returned to the player
+   * @param {PKExternalCaptionObject} caption - caption to be added to the model
+   * @param {Array<TextTrack>} playerTextTracks - player text tracks array
+   * @param {Array<TextTrack>} newTextTracks - text track array that will be returned to the player
+   * @returns {void}
+   * @private
+   */
+  _maybeAddTrack(track: TextTrack, caption: PKExternalCaptionObject, playerTextTracks: Array<Track>, newTextTracks: Array<TextTrack>): void {
```
Comment (dan-ziv) [inline src/track/external-captions-handler.js]: playerTextTracks: Array<TextTrack>

File: src/track/external-captions-handler.js (hunk 1 above)
Comment (odedhutzler) [reply src/track/external-captions-handler.js]: fixed

File: src/track/external-captions-handler.js (hunk 2 above)
Comment (odedhutzler) [reply src/track/external-captions-handler.js]: added a key, i think this with the error code which refers to the configuration docs is clear, no?

File: src/track/external-captions-handler.js (hunk 3 above)
Comment (odedhutzler) [reply src/track/external-captions-handler.js]: @dan-ziv don't understand what's the issue?

Comment (odedhutzler) [reply src/track/external-captions-handler.js]: flow doesn't consider filter, leaving it as Array<Track>

**Pull request merged**